lib/monitoring/addon/cluster-setting/route.js

```diff
@@ -1,11 +1,46 @@
 import Route from '@ember/routing/route';
-import { set } from '@ember/object';
+import { get, set } from '@ember/object';
 import { inject as service } from '@ember/service';
 import { on } from '@ember/object/evented';
 import C from 'ui/utils/constants';
+import { hash } from 'rsvp';
 
 export default Route.extend({
   session:     service(),
+  scope:       service(),
+  globalStore: service(),
+
+  model() {
+    const store = get(this, 'globalStore');
+
+    const cluster = get(this, 'scope.currentCluster');
+    const project = get(cluster, 'systemProject');
+
+    if ( !project || !get(cluster, 'enableClusterMonitoring') ) {
+      return { apps: [], }
+    }
+
+    return hash({
+      apps: store.rawRequest({
+        url:    `/v3/project/${ get(project, 'id') }/apps`,
```

Comment:
Is there a reason this can not be fetched using the project store?

Reply:
It's in cluster level menu, so no project is selected yet. I need to use the project store of System project in the cluster.

It looks like we can change the baseUrl if we want to use project store. Does it make sense?

```
    const projectStore = get(this, 'store')

    set(projectStore, 'baseUrl', `${ get(this, 'app.apiEndpoint') }/projects/${ project.id }`);
    const apps = projectStore.find('app', null, { forceReload: true })
```

Reply:
I'm sorry you changed it but with your explanation the way you disigned it originally makes more sense. It was not clear why that needed to be done that way though. Perhaps a small inline comment for any maintainers as to why we are using that would be helpful? I don't think changing the `baseUrl` is the right move though. That would change it for everything in the project store, which when you select a project should get fixed but we potentially introduce more issues. Id rather go back to rawRequest because now it makes sense.

Reply:
@westlywright No worries! But if we use rawRequest, I have another problem.

`get(app, 'externalIdInfo'); ` will return undefined as `externalIdInfo` field is in `models/app.js`.

How can I convert the json object to a model if I use rawRequest?

Reply:
Could you use the `_add(type, obj)` method on the store?

Reply:
Thanks! All I need is `createRecord`